tests/base_test_mixins.py

```diff
@@ -445,6 +596,63 @@ def assertGradeHistoryItemsCount(  # noqa
         result = json.loads(resp.content.decode())["result"]
         self.assertEqual(len(result), expected_count)
 
+    def get_update_course_url(self):
+        return reverse("relate-update_course", args=[self.course.identifier])
+
+    def temporarily_switch_user(self, target_user):
+        logged_in_user_changed = False
+        try:
+            logged_in_user_id = self.c.session['_auth_user_id']
+            from django.contrib.auth import get_user_model
+            logged_in_user = get_user_model().objects.get(
+                pk=int(logged_in_user_id))
+        except KeyError:
+            logged_in_user = None
+
+        if not target_user:
+            return logged_in_user, False
+
+        if logged_in_user != target_user:
+            self.c.force_login(target_user)
+            logged_in_user_changed = True
+
+        return logged_in_user, logged_in_user_changed
```

Comment:
I suppose I'd be more comfortable with this as a context manager, to make sure cleanup actually happens.

Reply:
I had that idea too, since you suggested,  I'll do so :)